content/blog/2020/06/2020-06-28-WinSW-yaml-support.adoc

```diff
@@ -0,0 +1,113 @@
+---
+layout: post
+title: "Windows Service Wrapprt : YAML Configuration Support - GSoC Phase - 01 Updates"
+tags:
+- winsw
+- windows
+- jenkins
+- gsoc
+- gsoc2020
+author: buddhikac96
+description: "This blog post discuss about phase 01 updates of Windows Service Wrapper Yaml support project of GSoC - 2020"
+opengraph:
+  image: /images/gsoc/opengraph.png
+---
+
+
+=== Introduction
+
+On Windows machines, Jenkins server and agents can be installed as Windows Services to get better robustness and manageability within the system. This is a functionality bundled into the Jenkins core directly. When installed as a service, Jenkins uses the Windows Service Wrapper executable (.NET, written in C#) which is being configured by XML config files. Currently, there are only a few configuration checks there (no XML Schema, limited validation, etc.), and it’s often that the service wrapper is misconfigured by Jenkins users.
+
+In GSoC-2020 what we are doing is update Windows service wrapper for YAML configurations support.
+
+=== Phase - 01 Updates
+
+- New CLI
+- YAML Configurations Support
+- XML Schema validation
+
+These updates are at the developping stage and not released yet. Hope to release all updates in GSoC-2020 Phase-02.
+
+=== New CLI
+[%hardbreaks]
+In the old CLI, it was in only verbs manner and the main update of the new CLI is converted CLI into verbs and option manner.
+So that users can specify the configuration file in the command line. 
+For command-line parsing, we used https://github.com/commandlineparser/commandline[Commandline] library.
+
+==== Available Options
+
+These options are common for all commands
+
+* --configFile <configuration_file>
+** User can specify the path of the configuration file. The current configuration file could be XML or YAML. If the config file is not specified WinSW will use the XML config file with the executable name. ex - winsw.xml
+
+* --skipConfigValidation <true/ false>
+** Users can skip the schema validation. If the user not specified WinSW will validate the configurations schema before deserializing. 
+
+* --redirect <redirect_path>
+** User can specify the path that Logs to be redirected
+
+* --elevated <true / false>
+** Elevate the command line
+
+* --help
+** Show available commands and options
+
+* --version
+** Show WinSW Version
+
+Some options are unique for commands
+
+* -p / --profile <true / false>
+** This option is available with install command and if the option is true then the user can specify the service account and allowing logon right in the command line
+
+==== Sample Command
+
+`winsw install --elevated true --confiFile "C:\Program Files\jenkins\config.yml" --skipConfigValidation true --profile true`
+
+
+=== YAML Configuration Support
+[%hardbreaks]
+YAML configurations support is under developing and after finalizing the YAML support users can specify configurations as a YAML file.
```

Comment:
`YAML configurations support is under developing` should be `YAML configurations support is under development`